app/controllers/journals_controller.rb

```diff
@@ -51,8 +48,12 @@ def journal_relations
   end
 
   def set_journal
-    @journal = Journal.find(params[:id])
-    authorize @journal
+    @journal = Journal.find_by(id: params[:id])
+    if @journal
+      authorize @journal
+    else
+      redirect_to set_journaled, notice: t('crud.c_action.destroy', model: t_model)
```

Comment:
Looks like `set_journaled` is called before each action, so you can just do `redirect_to @journaled, ...`

Reply:
For some reason, it is not called. I get `Cannot redirect to nil!` error. Maybe because it is a non route action?

Reply:
Ah sorry didn't realize this was inside `set_journal` :) It might work if you move `before_action :set_journaled` up so it's before `:set_journal`, they should be called in the order they're defined.